src/lib/drupal/lovell/utils.ts

```diff
@@ -133,3 +133,34 @@ export function isLovellBifurcatedResource(
     isLovellFederalResource(resource as LovellBifurcatedFormattedResource)
   )
 }
+
+export function getLovellVariantOfMenu(menu, variant: LovellVariant) {
+  if (!menu || typeof menu !== 'object') return menu
+
+  // Check if the object has the fieldMenuSection and it doesn't match the variant or is 'both'
+  if (
+    menu.fieldMenuSection &&
+    menu.fieldMenuSection !== variant &&
+    menu.fieldMenuSection !== 'both'
+  ) {
+    return null
+  }
+
+  // Filter arrays
+  if (Array.isArray(menu)) {
+    return menu
+      .map((item) => getLovellVariantOfMenu(item, variant))
+      .filter((item) => item !== null)
+  }
+
+  // Filter objects
+  const newData = {}
+  for (const key in menu) {
+    const filteredData = getLovellVariantOfMenu(menu[key], variant)
+    if (filteredData !== null) {
+      newData[key] = filteredData
+    }
+  }
```

Comment:
I'm not sure about this. Let's sync up and discuss some things.

Reply:
I understand what's going on here, but it feels a bit weird to me. It took me a bit to wrap my head around why there's "Filter arrays" and "Filter objects". I think that kind of muddies the situation a bit. It doesn't seem like we'd ever need to worry about objects separately, as the recursion should always be on the `links` property. 

The following code might not be perfect, as it depends on some of the other restructuring that we discussed, but it seems to me that this function should look something like this (general idea):

```
  return menu
    .filter(
      (menuItem) =>
        menuItem.fieldMenuSection === variant ||
        menuItem.fieldMenuSection === 'both'
    )
    .map((menuItem) => ({
      ...menuItem,
      links: getLovellVariantOfMenu(menuItem.links, variant),
    }))
```

Reply:
Updated this logic for clarify, thanks for the feedback